database/migrations/2023_03_24_113707_create_ab__article__has__article_categories_table.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('ab_article_has_articlecategory', function (Blueprint $table) {
+            $table->unsignedBigInteger('id')->primary();
+            $table->unsignedBigInteger('ab_articlecategory_id')->nullable(false);
+            $table->unsignedBigInteger('ab_article_id')->nullable(false);
+            $table->foreign('ab_articlecategory_id')->references('id')->on('ab_articlecategory');
+            $table->foreign('ab_article_id')->references('id')->on('ab_article');
+            $table->timestamps();
```

Comment:
$table->unique(['...', '...']); is missing. The combination of ab_articlecategory_id and ab_article_id is unique.